src/TimeSelect.jsx

```diff
@@ -111,10 +141,67 @@ var TimePicker = React.createClass({
     return this.generateFormattedTime((this.props.value.getHours() * 100) + this.props.value.getMinutes());
   },
 
-  render: function() {
+  hoursChanged: function(e) {
+    this.props.minutesHoursChanged({ hours: e.target.value });
+  },
+
+  minutesChanged: function(e) {
+    this.props.minutesHoursChanged({ minutes: e.target.value });
+  },
+
+  generateTimeInput: function(timeUnit) {
```

Comment:
this should probably be a separate component of TimeInput which accepts a label, currentTime and onChange handler

You could make a <TimeInput /> component and then extend this with <MinuteInput /> and <Hour /> input which pass in the timevalues and the label and accepts onChange and currentTime as props. Then when you're rendering this below, instead of 
```
<div className="row">
   {this.generateTimeInput('hours')}
   {this.generateTimeInput('minutes')}
</div>
```

you'd have
```
<div className="row">
  <HourInput  currentTime={new Date()} onChange={this.props.onChange} />
  <MinuteInput currentTime={new Date()} onChange={this.props.onChange} />
</div>
```